### PR TITLE
#159053956 Add descriptive signup validation errors

### PR DESCRIPTION
**Pull request by ronaldndirangu**
#### What does this PR do?
Add validation error messages for user signup

#### Description of tasks to be completed
Ensure that user gets descriptive error messages during signup in the case of any errors.

#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans.git
```
```sh
$ git checkout -b ft-descriptive-validation-errors-159053956
```
```sh
$ cd ah-titans/authors
```

> Add **.env** file and copy **example.env** content.
> Add your local database configuration settings to your **.env** file

```sh
$ source .env
```
```sh
$ python3.6 -m venv venv
```
```sh
$ source venv/bin/activate
```
```sh
$ pip install -r requirements.txt
```
```sh
$ python manage.py makemigrations
```
```sh
$ python manage.py test authentication
```

#### Any background context you want to provide?
No

#### What are the relevant Pivotal Tracker Stories?
[#159053956](https://www.pivotaltracker.com/story/show/159053956)

### CODE REVIEW

File: authors/tests/test_signup_validators.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            format='json'
+        )
+
+        self.assertIn("A password must contain atleast one small letter and one capital letter.",
+                      response.content.decode())
+        self.assertEquals(response.status_code,
+                          status.HTTP_400_BAD_REQUEST)
+
+    def test_for_capital_letter(self):
+        """Test user cannot signup if password lacks small letter"""
```
Comment (petermwash) [inline authors/tests/test_signup_validators.py]: Your doc strings on line 81 and 94 are the same and yet they are for different tests. Same as line 119 and 140.

Comment (ronaldndirangu) [reply authors/tests/test_signup_validators.py]: Thanks for pointing that out. I will make the appropriate changes.

**Comment by janetnim** [comment]
you've tested most edge cases so that's good.

File: authors/tests/test_signup_validators.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        self.capital_letter_password = {
+            "user": {
+                "username": "test",
+                "email": "test@gmail.com",
+                "password": "1234abcd"
+            }
+        }
+
+        """User data with invalid email"""
+        self.invalid_password = {
```
Comment (evansmusomi) [inline authors/tests/test_signup_validators.py:52]: The comment on line 51 and the variable name are confusing since one calls out the password as invalid while the other calls out the email as invalid. The object defined seems to have an invalid email with a valid password. In future, please use clear variable names.
